### PR TITLE
Moving internal funcitons to private scope

### PR DESCRIPTION
**Pull request by mlunoe**
This follows how `lodash` implements these functions

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @rcorral

**Comment by kennyt** [comment]
:ship:

**Comment by mlunoe** [comment]
@kennyt I did some restructuring to follow `lodash` completely. I also added documentation from `lodash`

File: src/Util/Util.js [hunk 1]
```diff
@@ -1,5 +1,339 @@
...
+
+/**
+ * Appends the elements of `items` to `array`.
+ *
+ * @private
+ * @param {Array} array The array to modify.
+ * @param {Array} items The items to append.
+ * @returns {Array} Returns `array`.
+ */
+function arrayPush(array, items) {
```
Comment (mlunoe) [inline src/Util/Util.js]: I changed `values` to `items` here, due to an eslint error

File: src/Util/Util.js [hunk 2]
```diff
@@ -1,5 +1,339 @@
...
+ * flattening and specifying the start index.
+ *
+ * @private
+ * @param {Array} array The array to flatten.
+ * @param {boolean} [isDeep] Specify a deep flatten.
+ * @param {boolean} [isStrict] Restrict flattening to arrays-like objects.
+ * @param {Array} [result=[]] The initial result value.
+ * @returns {Array} Returns the new flattened array.
+ */
+function baseFlatten(array, isDeep, isStrict, result) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 3]
```diff
@@ -1,5 +1,339 @@
...
+/**
+ * The base implementation of `_.pick` without support for individual
+ * property names.
+ *
+ * @private
+ * @param {Object} object The source object.
+ * @param {string[]} props The property names to pick.
+ * @returns {Object} Returns the new object.
+ */
+function basePick(object, props) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 4]
```diff
@@ -1,5 +1,339 @@
...
+ * The base implementation of `_.values` and `_.valuesIn` which creates an
+ * array of `object` property values corresponding to the property names
+ * of `props`.
+ *
+ * @private
+ * @param {Object} object The object to query.
+ * @param {Array} props The property names to get values for.
+ * @returns {Object} Returns the array of property values.
+ */
+function baseValues(object, props) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 5]
```diff
@@ -1,5 +1,339 @@
...
+}
+
+/**
+ * Excludes given properties from object
+ *
+ * @param  {Object} object
+ * @param  {Array} props Array of properties to remove
+ * @return {Object} New object without given props
+ */
+function exclude(object, props) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs.

File: src/Util/Util.js [hunk 6]
```diff
@@ -1,5 +1,339 @@
...
+ * _.isArrayLike(document.body.children);
+ * // => true
+ *
+ * _.isArrayLike('abc');
+ * // => true
+ *
+ * _.isArrayLike(_.noop);
+ * // => false
+ */
+function isArrayLike(value) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 7]
```diff
@@ -1,5 +1,339 @@
...
+ * _.isObjectLike([1, 2, 3]);
+ * // => true
+ *
+ * _.isObjectLike(_.noop);
+ * // => false
+ *
+ * _.isObjectLike(null);
+ * // => false
+ */
+function isObjectLike(value) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 8]
```diff
@@ -1,5 +1,339 @@
...
+ * @returns {boolean} Returns `true` if `value` is correctly classified, else `false`.
+ * @example
+ *
+ * _.isArguments(function() { return arguments; }());
+ * // => true
+ *
+ * _.isArguments([1, 2, 3]);
+ * // => false
+ */
+function isArguments(value) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 9]
```diff
@@ -1,5 +1,339 @@
...
+ * @returns {boolean} Returns `true` if `value` is correctly classified, else `false`.
+ * @example
+ *
+ * _.isFunction(_);
+ * // => true
+ *
+ * _.isFunction(/abc/);
+ * // => false
+ */
+function isFunction(value) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 10]
```diff
@@ -1,5 +1,337 @@
...
+ *  individually or in arrays.
+ * @returns {Object} Returns the new object.
+ * @example
+ *
+ * var object = { 'user': 'fred', 'age': 40 };
+ *
+ * _.pick(object, 'user');
+ * // => { 'user': 'fred' }
+ */
+function pick(object, props) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 11]
```diff
@@ -1,5 +1,337 @@
...
+ *
+ * Foo.prototype.c = 3;
+ *
+ * _.values(new Foo);
+ * // => [1, 2] (iteration order is not guaranteed)
+ *
+ * _.values('hi');
+ * // => ['h', 'i']
+ */
+function values(object) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Same function, just added docs

File: src/Util/Util.js [hunk 12]
```diff
@@ -27,6 +359,57 @@ function es6ify(mixin) {
...
+    }
+  });
+  return result;
+}
+
+function noop() {
+  return null;
+}
+
+function sortBy(collection, sortProp) {
```
Comment (rcorral) [inline src/Util/Util.js]: Is this the same?

File: src/Util/Util.js [hunk 13]
```diff
@@ -1,5 +1,337 @@
...
+ * // => [1, 2] (iteration order is not guaranteed)
+ *
+ * _.values('hi');
+ * // => ['h', 'i']
+ */
+function values(object) {
+  return object ? baseValues(object, Object.keys(object)) : [];
+}
+
+// Other functions
```
Comment (kennyt) [inline src/Util/Util.js]: what does `Other functions` mean?

File: src/Util/Util.js [hunk 14]
```diff
@@ -1,5 +1,337 @@
...
+  return newObject;
+}
+
+/**
+ * Checks if `value` is array-like. A value is considered array-like if it's
+ * not a function and has a `value.length` that's an integer greater than or
+ * equal to `0` and less than or equal to `Number.MAX_SAFE_INTEGER`.
+ *
+ * @static
+ * @memberOf _
```
Comment (kennyt) [inline src/Util/Util.js]: it's no longer a `@memberOf _`

File: src/Util/Util.js (hunk 12 above)
Comment (rcorral) [inline src/Util/Util.js]: It is...

**Comment by rcorral** [comment]
:shipit: after @kennyt comments are addressed.

File: src/Util/Util.js (hunk 14 above)
Comment (mlunoe) [inline src/Util/Util.js]: Removed.

**Pull request merged**